ingestion/src/metadata/profiler/metrics/static/max.py

```diff
@@ -43,6 +44,15 @@ def _(element, compiler, **kw):
 @compiles(MaxFn, Dialects.Trino)
 def _(element, compiler, **kw):
     col = compiler.process(element.clauses, **kw)
+    first_clause = element.clauses.clauses[0]
+    # Check if the first clause is an instance of LenFn and its type is not in FLOAT_SET
+    # or if the type of the first clause is date time
+    if (
+        isinstance(first_clause, LenFn)
+        and type(first_clause.clauses.clauses[0].type) not in FLOAT_SET
+    ) or is_date_time(first_clause.type):
+        # If the condition is true, return the maximum value of the column
+        return f"MAX({col})"
```

Comment:
can't we just check the type of `col` instead of that `isinstance(...)`

Reply:
we send either a column or in other cases a LenFn with a column as argument, on check type, it shows type as NullType if we do first_clause.type for LenFn